src/main/java/net/geforcemods/securitycraft/entity/SentryEntity.java

```diff
@@ -224,6 +237,19 @@ else if(item == SCContent.UNIVERSAL_OWNER_CHANGER.get())
 	@Override
 	public void remove()
 	{
+		BlockPos pos = getPosition();
+		if (!getDisguiseModule().isEmpty())
+		{
+			List<ItemStack> blocks = ((ModuleItem)getDisguiseModule().getItem()).getAddons(getDisguiseModule().getTag());
+			if(blocks.size() > 0)
+			{
+				BlockState state = Block.getBlockFromItem(blocks.get(0).getItem()).getDefaultState();
+				if(state != world.getBlockState(pos))
+					Block.spawnAsEntity(world, getPosition(), new ItemStack(world.getBlockState(pos).getBlock().asItem()));
+			}
+		}
+		else Block.spawnAsEntity(world, pos, new ItemStack(world.getBlockState(pos).getBlock().asItem()));

```

Comment:
The logic here should be that it removes the block if it matches up with the one in the disguise module, and doesn't do anything otherwise. Once you change that, don't forget to update the comment above this method as it no longer removes any block after that change.

Reply:
Sorry, but I don't get your point, because:
> logic should be that it removes the block if ...

and
> it no longer removes any block after that change

Reply:
Bad wording on my part. I mean that it no longer removes the block no matter which one, but only some blocks.